src/main/java/org/betonquest/betonquest/compatibility/mythicmobs/MythicMobKillObjective.java

```diff
@@ -68,18 +68,18 @@ public void onBossKill(final MythicMobDeathEvent event) {
         if (!names.contains(event.getMobType().getInternalName())) {
             return;
         }
-        if (marked != null) {
-            if (!event.getEntity().hasMetadata("betonquest-marked")) {
-                return;
-            }
-            final List<MetadataValue> meta = event.getEntity().getMetadata("betonquest-marked");
-            for (final MetadataValue m : meta) {
-                if (!m.asString().equals(marked)) {
+        if (event.getKiller() instanceof Player) {
+            if (marked != null) {
+                if (!event.getEntity().hasMetadata("betonquest-marked")) {
                     return;
                 }
+                final List<MetadataValue> meta = event.getEntity().getMetadata("betonquest-marked");
+                for (final MetadataValue m : meta) {
+                    if (!m.asString().equals(marked.replace("%player%", event.getKiller().getName()))) {
+                        return;
+                    }
+                }
             }
```

Comment:
this change introduce a new bug:
If a mob is killed, and no killer is given, it counts now the objective anyway (neutralDeathRadiusAllPlayers) but foe all players. I thing this change it self is ok, but below in the else part you need need to resolve the mearked attribute for every player in that radius (may in line 90)